toolchain/defs.bzl

```diff
@@ -213,12 +234,21 @@ def _zig_repository_impl(repository_ctx):
     if not zig_wrapper_success:
         fail(zig_wrapper_err_msg)
 
-    exe = ".exe" if os == "windows" else ""
+    exe = ".exe" if exec_os == "windows" else ""
     for t in _BUILTIN_TOOLS:
         repository_ctx.symlink("tools/zig-wrapper{}".format(exe), "tools/{}{}".format(t, exe))
 
+    urls = [uf.format(**format_vars_exec) for uf in repository_ctx.attr.url_formats]
+
+    repository_ctx.download_and_extract(
```

Comment:
Something to discuss since you will understand the flow better than me; this is my "hack" for getting the exec toolchain in place since we need the host toolchain to build the zig-wrapper.

I believe this is fine as zig is small enough; however it isn't ideal. I need to think about if there is a way to build the `zig-wrapper` executable outside of the repository_ctx or not; otherwise maybe having a `@host-zig_sdk` could work, but I worry about how to keep the cache hit rate good if we're using non-relative paths + we'd have to ensure that repository is local, which I do think the `local` parameter will handle for us, but I need to think more on it.

Reply:
yeah, I don't see a way to avoid downloading two Zig SDKs when host and exec platforms are different. Building zig-wrapper in repository rules is an intentional design to read `HERMETIC_CC_TOOLCHAIN_CACHE_PREFIX`, which is not possible during regular build actions.